packages/core/src/components/axes/zoom-bar.ts

```diff
@@ -33,6 +33,11 @@ export class ZoomBar extends Component {
 	xScale: any;
 	yScale: any;
 
+	// keep the initial zoomBar domain to avoid the incorrect domain due to data changes
+	defaultZoomBarDomain: any;
+
+	zoomService = this.services.zoom;
```

Comment:
we shouldn't do this

Reply:
Updated in https://github.com/carbon-design-system/carbon-charts/pull/717/commits/1e6ff2aca938073a1ddbd96fd979a27d72525026.